pep-0613.rst

```diff
@@ -0,0 +1,173 @@
+
+::
+
+  x: TypeAlias = ClassName
+  def foo() -> None:
+    x = ClassName
+    x: TypeAlias = ClassName
+
+With explicit aliases, the outer assignment is still a valid type variable, and the inner assignment can either be a valid local variable or a clear error, communicating to the author that type aliases cannot be defined inside a nested scope.
+
+
+Specification
+=============
+
+The explicit alias declaration syntax attempts to clearly differentiate between the three possible kinds of assignments: typed global expressions, untyped global expressions, and type aliases. This avoids the existence of assignments that break type checking when an annotation is added, and avoids classifying the nature of the assignment based on the type of the value.
+
+Implicit syntax (pre-existing):
+
+::
+
+  x = 1  # untyped global expression
+  x: int = 1  # typed global expression
+
+  x = int  # type alias
+  x: Type[int] = int  # typed global expression
+
+
+Explicit syntax:
+
+::
+
+  x = 1  # untyped global expression
+  x: int = 1  # typed global expression
+
+  x = int  # untyped global expression (see note below)
+  x: Type[int] = int  # typed global expression
+
+  x: TypeAlias = int  # type alias
+  x: TypeAlias = “MyClass”  # type alias
+
+
+Note: The examples above illustrate implicit and explicit alias declarations in isolation. For the sake of backwards compatibility, type checkers may choose to support both simultaneously, meaning an untyped global expression :code:`x = int` will still be considered a valid type alias.
+
+
+Backwards Compatibility
+=======================
+
+Explicit aliases provide an alternative way to declare type aliases, but all pre-existing code and old alias declarations will work as before.
+
+
+Reference Implementation
+========================
+
+The `Pyre <https://pyre-check.org/>`_ type checker supports explicit type alias declarations.
+
+
+Rejected Ideas
+==============
+
+Some alternative syntaxes were considered for explicit aliases:
+
+::
+
+  MyType: TypeAlias[int]
+
+This looks a lot like an uninitialized variable.
+
+::
+
+  MyType = TypeAlias[int]
+
+Along with the option above, this format potentially adds confusion around what the runtime value of :code:`MyType` is.
+
+
+In comparison, the chosen syntax option :code:`MyType: TypeAlias = int` is appealing because it still sticks with the :code:`MyType = int` assignment syntax, and adds some information for the type checker purely as an annotation.
+
+
+Open Issues
+===========
+
+
+The current implementation utilizes the already-defined class :code:`typing.TypeAlias`, but for conciseness there are suggestions to standardize this to a new class, :code:`typing.Alias`.
```

Comment:
I don't see `typing.TypeAlias` in any of the versions of `typing.py` that I'm aware of. And I'm guess the open issue is whether to use `TypeAlias` or just `Alias`? I vote for `TypeAlias`, because "alias" can mean many things in many contexts.

Reply:
Hm, I meant it's defined in typing.pyi, and so it's possible already to write `from typing import TypeAlias` and use it in your annotations. I suppose this shouldn't matter in versions where annotations aren't evaluated, but let me try to clarify this wording and note that it should probably be added to typing.py. I'd also vote for TypeAlias, but remember that I got more than one suggestion to shorten from others on the list/at the meetup.